rpc/common/moc_common_networkcommon.pb.go

```diff
@@ -351,6 +351,53 @@ func (m *Route) GetMetric() uint32 {
 	return 0
 }
 
+type NsgGroup struct {
```

Comment:
>NsgGroup [](http://example.com/codeflow?start=5&length=8)

why cant we use the existing NetworkSecurityGroup ?

Reply:
We could, but I wanted it to be clear that this is a stub. I can change it if you'd prefer.

Reply:
yeah, let's use the existing one. it's a well-established pattern in both networkcontroller and in azure.

Reply:
thinking abt it, there is another problem. when we introduce back references, ideally, we would want the NSG proto to have a stub LSubnet resource. but such circular dependencies are not supported in protobuf.
one option is, for the read/write ref we will use a real typed stub resource, like in this case. but for readonly ref (like the one in NSG), we will introduce a new ResourceReference type and use that. 

but with that approach we will be restricting ourselves for the future. there are some scenarios like SLB where a backend pool is read/writeable from both nic and lb.

What about we introduce a common ResourceReference type (similar to moc-sdk-for-go) and use that? 

``` go
ResourceReference
     Location string (optional)
     Group string
     Name string
```

Reply:
implemented!

Reply:
thank you doing this!

Reply:
@raghavendra-nataraj fyi